classifier_and_summarizer/summarizer/extractor_output_preprocessor.py

```diff
@@ -0,0 +1,219 @@
+
+    def _summarize_text_content(self):
+        '''
+        Applies text summarization to the combined text in the webpage
+        '''
+        # aggregate all text from the webpage
+        webpage_text = '. '.join(
+            [text.text_string for text in self.normal_text_content_list])
+        # sentence tokenize to get list of summary sentences
+        self.summarized_text = sent_tokenize(
+            self.text_summarizer.summarize_text(webpage_text))
+
+    def _get_sentence_object(self, summarized_text_index, normal_text_index):
+        ''' instantiates/initializes and returns a sentence object'''
+        return SentenceWithAttributes(
+            self.summarized_text[summarized_text_index],
+
+            self.normal_text_content_list[
+                normal_text_index].content_index,
+
+            self.normal_text_content_list[
+                normal_text_index].font_style,
+
+            self.summarized_text_embeddings[
+                summarized_text_index]
+        )
+
+    def _text_object_has_sentence(
+            self,
+            summarized_text_index,
+            normal_text_index):
+        '''
+        Given a summarized sentence checks if the given
+        text object at that index contains the sentence
+        '''
+        return self.summarized_text[summarized_text_index] \
+            in self.normal_text_content_list[normal_text_index].text_string
+
+    def _make_sentence_objects(self):
+        ''' Assembles the sentence objects with required attributes
+
+        Note that the summarized text is a subsequence of the
+        webpage text. We use this fact to assemble the sentence
+        objects
+
+        Maintain two running indices through each list
+        i: running index in normal text content
+        j: running index in summarized text content
+        - When the text is present at some indices i,j in the lists we
+            increment both i and j
+        - Else just increment i
+        '''
+        self.sentence_objects_list = list()
+        self.running_index_in_normal_text_content = 0
+        self.running_index_in_summarized_text = 0
+
+        # encode all sentences together - reduces latency
+        self.summarized_text_embeddings = self.sentence_embedding_model.encode(
+            self.summarized_text)
+
+        count_of_summary_sentences = len(self.summarized_text)
+
+        while self.running_index_in_summarized_text \
+                < count_of_summary_sentences:
+
+            if self._text_object_has_sentence(
+                self.running_index_in_summarized_text,
+                self.running_index_in_normal_text_content
+            ):
+
+                self.sentence_objects_list.append(
+                    self._get_sentence_object(
+                        self.running_index_in_summarized_text,
+                        self.running_index_in_normal_text_content
+                    )
+                )
+                self.running_index_in_summarized_text += 1
```

Comment:
self.running_index_in_normal_text_content += 1 also here?

Reply:
It is possible that two summarized sentences could lie in say perhaps one large paragraph - that is why `self.running_index_in_normal_text_content` is not incremented here (in case the next summarized sentence also lies in the same paragraph) and is only incremented when the sentence isn't found in the block of text

Reply:
Makes sense! Thanks!